pkg/ssh/connection.go

```diff
@@ -273,31 +271,39 @@ func (c *connection) Close() error {
 	return c.sshclient.Close()
 }
 
-func (c *connection) Stream(cmd string, stdout io.Writer, stderr io.Writer) (int, error) {
+func (c *connection) POpen(cmd string, stdin io.Reader, stdout io.Writer, stderr io.Writer) (int, error) {
 	sess, err := c.session()
 	if err != nil {
 		return 0, errors.Wrap(err, "failed to get SSH session")
 	}
 	defer sess.Close()
 
+	sess.Stdin = stdin
 	sess.Stdout = stdout
 	sess.Stderr = stderr
 
 	exitCode := 0
-	err = sess.Run(strings.TrimSpace(cmd))
-	if err != nil {
-		exitCode = 1
+	if err = sess.Run(cmd); err != nil {
+		exitCode = -1
```

Comment:
Why we use exit code `-1`? Negative exit codes are not really a thing, so this might be confusing.